solutions/usaco-572.mdx

```diff
@@ -52,55 +57,94 @@ int main() {
 ### Implementation
 
 ```java
-import java.io.File;
-import java.io.PrintWriter;
-import java.util.Scanner;
-
-public class bcount {
-	public static void main(String[] args) throws Exception {
-		Scanner sc = new Scanner(new File("bcount.in"));
-		PrintWriter out = new PrintWriter("bcount.out");
-
-		int N = sc.nextInt();
-		int Q = sc.nextInt();
-		int[] Holsteins = new int[N + 1];
-		int[] Guernseys = new int[N + 1];
-		int[] Jerseys = new int[N + 1];
-
-		for (int i = 1; i < N + 1; i++) {
-			Holsteins[i] += Holsteins[i - 1];
-			Guernseys[i] += Guernseys[i - 1];
-			Jerseys[i] += Jerseys[i - 1];
-
-			int breed = sc.nextInt();
-
-			if (breed == 1) {
-				Holsteins[i] ++;
-			} else if (breed == 2) {
-				Guernseys[i] ++;
-			} else if (breed == 3) {
-				Jerseys[i] ++;
-			} else {
-				System.out.println("What the...");
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class BCount {
+	public static void main(String[] args) throws IOException {
+		BufferedReader read = new BufferedReader(new FileReader("bcount.in"));
+		StringTokenizer initial = new StringTokenizer(read.readLine());
+		int cowNum = Integer.parseInt(initial.nextToken());
+		int queryNum = Integer.parseInt(initial.nextToken());
+
+		int[] holsteins = new int[cowNum + 1];
+		int[] guernseys = new int[cowNum + 1];
+		int[] jerseys = new int[cowNum + 1];
+		for (int i = 1; i <= cowNum; i++) {
+			holsteins[i] += holsteins[i - 1];
+			guernseys[i] += guernseys[i - 1];
+			jerseys[i] += jerseys[i - 1];
+
+			int breed = Integer.parseInt(read.readLine());
+			switch (breed) {
+				case 1:
+					holsteins[i]++;
+					break;
+				case 2:
+					guernseys[i]++;
+					break;
+				case 3:
+					jerseys[i]++;
+					break;
+				default:
+					throw new IllegalArgumentException("invalid cow breed given");
 			}
 		}
 
-		for (int i = 0; i < Q; i++) {
-			int start = sc.nextInt();
-			int end = sc.nextInt();
-
-			int Holstein = Holsteins[end] - Holsteins[start - 1];
-			int Guernsey = Guernseys[end] - Guernseys[start - 1];
-			int Jersey = Jerseys[end] - Jerseys[start - 1];
-
-			out.println(Holstein + " " + Guernsey + " " + Jersey);
+		StringBuilder ans = new StringBuilder();
```

Comment:
no point in using a stringbuilder - just print out the answer.

Reply:
if i don't use a stringbuilder it TLEs

Reply:
it tles when you print it out?? Dont think that's possible

Reply:
io's pretty slow in java, just trust me on this one

Reply:
![Screen Shot 2021-05-17 at 1 07 04 PM](https://user-images.githubusercontent.com/59348162/118549791-c953bc80-b710-11eb-9e37-8db9aea22c73.png)

wdym?

Reply:
![image](https://user-images.githubusercontent.com/55369003/118552913-70862300-b714-11eb-97ae-37bf89770759.png)
idk man, maybe i'm just trolling but

Reply:
wait im p sure printing it out is always going to be faster than stringbuilder.

Reply:
^I could be wrong but nevertheless printing should always pass still

Reply:
i think it's cause @SansPapyrus683 was using printf?

Reply:
> wait im p sure printing it out is always going to be faster than stringbuilder.

most definitely not